src/workerd/util/sqlite.c++

```diff
@@ -30,6 +31,102 @@
 
 namespace workerd {
 
+namespace {
+
+// SQLite has a function like this in its internals, but it's not exposed to library consumers.
+//
+// These error codes come from https://www.sqlite.org/rescode.html#primary_result_code_list.
+kj::Maybe<kj::LiteralStringConst> errorName(int errorCode) {
+  switch (errorCode) {
+  case SQLITE_OK:
+    return "SQLITE_OK"_kjc;
+  case SQLITE_ERROR:
+    return "SQLITE_ERROR"_kjc;
+  case SQLITE_INTERNAL:
+    return "SQLITE_INTERNAL"_kjc;
+  case SQLITE_PERM:
+    return "SQLITE_PERM"_kjc;
+  case SQLITE_ABORT:
+    return "SQLITE_ABORT"_kjc;
+  case SQLITE_BUSY:
+    return "SQLITE_BUSY"_kjc;
```

Comment:
it would likely be a bit cleaner to use a macro expansion for this instead...

something like...

```cpp
  #define SQLITE_ERRORS(V) \
    V(SQLITE_OK) \
    V(SQLITE_ERROR) \
    ...

#define V(name) case name: return #name##_kj;
  switch (1) {
    SQLITE_ERRORS(V)
  }
#undef V
```

This would reduce the duplication and make things a bit more compact.

non-blocking tho

Reply:
Fair enough. I'm not a big fan of how macro-heavy our code is sometimes, but as long as it's scoped to just this function, it should be fine. Done in commit b15e6940.